### DISCUSSION TITLE
NetBSD package

### DISCUSSION THREAD

**Discussion by 0323pin** [spenserblack/repofetch#237] [Development]
FYI, I've packaged `repofetch` for NetBSD and have merged the package into the main repo, https://mail-index.netbsd.org/pkgsrc-changes/2022/10/09/msg261795.html

Regards

**Comment by spenserblack** [comment]
Thanks for this! I'm not familiar with NetBSD, but from some quick skimming this means that repofetch is now installable with `pkgin`, right? Would you like to make a PR to update the installation docs?

**Comment by 0323pin** [comment]
Yes, I can make a PR and add it to the README.md.
Yes, you can install it with `pkgin` depending on which branch you are tracking. On current/development branch it should be available as a pre-compiled binary already tomorrow or the day after. The stable branch is released every three months, so there it should become available soon after Christmas.

I run the bleeding edge branch myself.

**Comment by spenserblack** [comment]
Hey, just an FYI that [I'm considering changing the implementing language](https://github.com/spenserblack/repofetch/issues/186#issue-1404594032). I haven't made any decision yet, but I'm assuming that if I did so it would severely affect the NetBSD package, wouldn't it?

**Comment by 0323pin** [comment]
@spenserblack Thanks for the heads-up. It really depends on what the changes look like. And, it's your app/work so, you decide.

If you move to python, js or, ruby I would ask if someone else is willing to take over as a package maintainer. I don't know anything about packaging stuff on those languages.
My packaging experience is limited to rust, c++, c and go.

**Comment by spenserblack** [comment]
@allcontributors please add @0323pin for platform

**Comment by allcontributors** [comment]
@spenserblack 

I've put up [a pull request](https://github.com/spenserblack/repofetch/pull/190) to add @0323pin! :tada:

**Comment by 0323pin** [comment]
@spenserblack Thanks, I'm honoured.

**Comment by spenserblack** [comment]
@0323pin Just an FYI that I've decided to keep the Rust version in the `rust` branch, and it may still receive updates.

**Reply by 0323pin** [reply]
I'll have a look at packaging the new `ruby` version during the X-tmas break.

**Reply by spenserblack** [reply]
Thank you! Let me know if you have any questions :heart:

**Reply by spenserblack** [reply]
Just checked and saw that [it looks like there's a NetBSD package for octokit.rb](https://pkgsrc.se/devel/ruby-octokit). Could be a good example package for Ruby projects. Though, unlike repofetch, octokit doesn't have an executable.

**Reply by 0323pin** [reply]
Would you have a list of all build and runtime dependencies available?

**Reply by spenserblack** [reply]
Here's a list of direct dependencies.

## Build

Only `gem`, which is part of the Ruby build tools, should be directly required. `gem build` to read the `.gemspec` file and build a `.gem` file, and `gem install` to read that `.gem` file and unpack it to the correct location. [`gem.mk`](https://github.com/NetBSD/pkgsrc/blob/trunk/lang/ruby/gem.mk) might be able to offer some more info. Technically, I don't think even `gem` is required if its behavior is mimicked by manually copying all files.

If docs are going to be built, then [`yard`](https://pkgsrc.se/textproc/ruby-yard) might be considered a build dependency.

## Runtime

- [`dotenv`](https://pkgsrc.se/devel/ruby-dotenv) (this seems to be a deleted pkgsrc package)
- [`git (Ruby gem)`](https://pkgsrc.se/devel/ruby-git)
- [`actionview`](https://pkgsrc.se/www/ruby-actionview70) (I'd eventually like to drop this dependency, because it feels like overkill for only a couple of utility functions)

### Optional (required by plugin)

- [`ruby-octokit`](https://pkgsrc.se/devel/ruby-octokit) Is optional, but highly recommended. In Rust terms, `default-features = ["octokit"]`
- [`faraday-retry`](https://rubygems.org/gems/faraday-retry) is optional, and honestly just added to silence a warning that octokit generates.

**Reply by 0323pin** [reply]
Awesome, thanks!
I wasn't expecting you to dig through pkgsrc to find availability 🤗 
Thanks for going the extra mile.

**Reply by spenserblack** [reply]
Just a heads up: #222 will change sawyer from an indirect dependency (via octokit) to a direct dependency.

**Reply by 0323pin** [reply]
Cheers! I was away for X-mas and didn't have time for creating a new package. I'll have time on Monday, will let you know if I _"hit the wall"_ 😄

**Reply by spenserblack** [reply]
No problem, enjoy your break!

**Comment by 0323pin** [comment]
@spenserblack Just FYI, I've created a skeleton package update today. Unfortunately, the default `ruby` version has been bumped to 3.1.3 which would force me to re-build the whole ruby infrastructure locally. I don't have the time or resources needed for this, so I need to wait for pre-compiled binaries to become available from the servers before I can test the build.

I'm a bit concerned with the missing dependencies, `ruby-dotenv` and `ruby-git` but, until I can get my hands on fresh `ruby` binaries, I won't be able to test any builds.

Sorry for the delay :(

**Reply by spenserblack** [reply]
>Sorry for the delay :(

No worries, there's no rush 🙂 

Thanks for the head up!

**Reply by spenserblack** [reply]
Just had a thought: would you know how frequently NetBSD's default Ruby version bumped? I test 3.1 in the CI, but I wonder if it would be worth it to test 3.1.3 explicitly for the NetBSD package.

**Reply by 0323pin** [reply]
> Just had a thought: would you know how frequently NetBSD's default Ruby version bumped? I test 3.1 in the CI, but I wonder if it would be worth it to test 3.1.3 explicitly for the NetBSD package.

Honestly, I have no clue how often the default Ruby version is bumped. Usually, I don't keep track of it as I mostly deal with Rust packages.

I've asked our Ruby expert for help and he has managed to get `repofetch` running. Although, he will need to add a few of the currently unavailable dependencies first. I'll ask nicely if he can update the `repofetch` package to the newest version.

**Reply by spenserblack** [reply]
Alright, thanks so much for working to get repofetch available on NetBSD!

**Comment by 0323pin** [comment]
@spenserblack Thanks to Takahiro Kambe, we now have the Ruby version :)
https://pkgsrc.se/sysutils/ruby-repofetch

**Reply by spenserblack** [reply]
Thank you so much! I appreciate all the hard work!
